### PR TITLE
Adjust Muzzle Devices

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
Content "Adjusts various muzzle devices"

#### Purpose of change
Suppressors were the wrong length. You could craft suppressors too easily, forming threads somehow out of thin air, or magicking baffles into existence. Finally we had some unnecessary suppressor models, the rivtech one that was somehow better than regular ones, and the soda bottle suppressor which is frankly pretty useless. You could easily craft a rifle rated suppressor, which is a bit more difficult than just a pistol one.

#### Describe the solution
Migrated offending suppressors.
Improved the recipes for all craftable suppressors - Swapped the recipe for the suppressor to instead produce the pistol can.
Renamed the compact suppressor to the pistol suppressor as it's more clear.
Added alternate recipes for craftable suppressors using pre-existing muzzle attachments to take a small shortcut.
Adjusted pistol suppressor description.

#### Describe alternatives you've considered
24.2 welding rods seems like a LOT of welding rods to fully weld a suppressor, especially one doing a  folded-sheetmetal construction. I tossed my conditions and calculated weld length into a weld calculator. 
image
This is only about 6welding rods. I considered lowering the recipe to this.

#### Testing
Loaded game. Tried making stuff. 

#### Additional context
None.

### CODE REVIEW

File: data/json/items/gunmod/muzzle.json [hunk 1]
```diff
@@ -223,12 +181,12 @@
   {
     "id": "suppressor_compact",
     "type": "GUNMOD",
-    "name": { "str": "compact suppressor" },
-    "description": "A compact suppressor designed for pistols and best used with smaller calibers.",
+    "name": { "str": "pistol suppressor" },
+    "description": "A suppressor designed for pistols, submachine guns, and pistol caliber carbines. Gunfire is extremely loud and can damage your hearing without protection; a suppressor will bring the loudness of a report down to generally safe levels.",
```
Comment (TheShadowFerret) [inline data/json/items/gunmod/muzzle.json:185]:
```suggestion
    "description": "A suppressor designed for pistols, submachine guns, and pistol caliber carbines.  Gunfire is extremely loud and can damage your hearing without protection; a suppressor will bring the loudness of a report down to generally safe levels.",
```

**Comment by Hyperseeker** [comment]
[The official store listing](https://silencerco.com/silencers/omega-k/) for the Omega K series lists 45's length as 6.37″ = 161.8 mm ≈ 162 mm. Other sources ([Silencer Warehouse](https://www.silencerwarehouse.com/product-p/sco-omega45k.htm), [Guns n Gear](https://gngsports.com/product/silencerco-omega-45k-45acp/)) list it as 6.36″ = 161.54 mm ≈ 162 mm. Out of the top 10 results on Google, the only one to list it at 6.4″ = 162.56 mm ≈ 163 mm is [Silencer Shop](https://www.silencershop.com/silencerco-omega-45k.html).

Salvo 12 [is stated to](https://silencerco.com/shop/salvo-12/) have the maximum length of 12.5″ = 317.5 mm ≈ 318 mm (or 12.52″ = 318.01 mm ≈ 318 mm, if you believe [Silencer Shop](https://www.silencercentral.com/products/salvo-12-gauge-shotgun-silencer-silencerco/)). It doesn't appear to be able to change length to slightly under 12″: the steps (however inaccurate) are stated to start with 10″.

Hybrid 46 DTM [comes down to](https://www.silencershop.com/silencerco-hybrid-46-dtm.html) 7.8″ = 198.12 mm ≈ 198 mm OAL.

**Comment by tenmillimaster** [comment]
> [The official store listing](https://silencerco.com/silencers/omega-k/) for the Omega K series lists 45's length as 6.37″ = 161.8 mm ≈ 162 mm. Other sources ([Silencer Warehouse](https://www.silencerwarehouse.com/product-p/sco-omega45k.htm), [Guns n Gear](https://gngsports.com/product/silencerco-omega-45k-45acp/)) list it as 6.36″ = 161.54 mm ≈ 162 mm. Out of the top 10 results on Google, the only one to list it at 6.4″ = 162.56 mm ≈ 163 mm is [Silencer Shop](https://www.silencershop.com/silencerco-omega-45k.html).
> 
> Salvo 12 [is stated to](https://silencerco.com/shop/salvo-12/) have the maximum length of 12.5″ = 317.5 mm ≈ 318 mm (or 12.52″ = 318.01 mm ≈ 318 mm, if you believe [Silencer Shop](https://www.silencercentral.com/products/salvo-12-gauge-shotgun-silencer-silencerco/)). It doesn't appear to be able to change length to slightly under 12″: the steps (however inaccurate) are stated to start with 10″.
> 
> Hybrid 46 DTM [comes down to](https://www.silencershop.com/silencerco-hybrid-46-dtm.html) 7.8″ = 198.12 mm ≈ 198 mm OAL.

I'm going to update the values again based on the stats in their manuals, here: <https://silencerco.com/wp-content/uploads/2020/09/Salvo_Manual-web.pdf> <https://silencerco.com/wp-content/uploads/2020/05/Hybrid_Manual.pdf> <https://silencerco.com/wp-content/uploads/2020/05/OmegaK_Manual.pdf>

**Comment by Cenness** [comment]
Would be nice if someone also adjusted holsters, so we could finally skip crafting big variants to be able to use them with suppressed handguns.

**Comment by tenmillimaster** [comment]
> Would be nice if someone also adjusted holsters, so we could finally skip crafting big variants to be able to use them with suppressed handguns.

You would have to do that regardless after any adjustment to existing holsters. You can't holster a pistol in a standard holster.

**Pull request merged**